src/main/java/io/cdap/plugin/splunk/common/client/SplunkSearchClient.java

```diff
@@ -0,0 +1,168 @@
+  }
+
+  /**
+   * Returns sample data for Splunk Search.
+   *
+   * @return List of events
+   * @throws IOException thrown if there are any issue with the I/O operations.
+   */
+  public List<Map<String, String>> getSample() throws IOException {
+    long count = 100L;
+    SplunkBatchSourceConfig configForSchema = getConfigForSchema("Normal");
+    String searchId = getSearchId(configForSchema, count);
+    List<Map<String, String>> sample = new ArrayList<>();
+    try (SplunkSearchIterator iterator = buildSearchIterator(searchId, 0L, count)) {
+      iterator.forEachRemaining(sample::add);
+    }
+    return sample;
+  }
+
+  /**
+   * Returns total results for Splunk search.
+   *
+   * @return Total results for search
+   * @throws IOException thrown if there are any issue with the I/O operations.
+   */
+  public long getTotalResults() throws IOException {
+    long count = 1L;
+    SplunkBatchSourceConfig configForSchema = getConfigForSchema("Blocking");
+    String searchId = getSearchId(configForSchema, count);
+    try (SplunkSearchIterator iterator = buildSearchIterator(searchId, 0L, count)) {
+      iterator.hasNext();
+    }
+    Job job = splunkService.getJob(searchId);
+    return job.getResultCount();
+  }
+
+  private SplunkBatchSourceConfig getConfigForSchema(String executionMode) {
+    return new SplunkBatchSourceConfig(config.referenceName,
+                                       config.getAuthenticationTypeString(),
+                                       config.getToken(),
+                                       config.getUsername(),
+                                       config.getConnectTimeout(),
+                                       config.getReadTimeout(),
+                                       config.getNumberOfRetries(),
+                                       config.getMaxRetryWait(),
+                                       config.getMaxRetryJitterWait(),
+                                       config.getUrlString(),
+                                       config.getPassword(),
+                                       executionMode,
+                                       config.getOutputFormat(),
+                                       config.getSearchString(),
+                                       config.getSearchId(),
+                                       config.getAutoCancel(),
+                                       config.getEarliestTime(),
+                                       config.getLatestTime(),
+                                       config.getIndexedEarliestTime(),
+                                       config.getIndexedLatestTime(),
+                                       config.getSearchResultsCount(),
+                                       config.getSchema());
+  }
+
+  private String getSearchId(SplunkBatchSourceConfig config, Long count) {
+    Job job = getJob(config);
+    while (!job.isDone()) {
+      if (!job.isReady()) {
```

Comment:
it will consume all CPU until ready, some sleep required.

Reply:
Done.